### PR TITLE
Fixes + Enhancements

### LINKED ISSUE #2529: Character (or query param) limit in HAPI query URL
**Describe the bug**
We have discovered that the GET query URL for StructureMaps with the id (UUID) list has a limit of approximately 35 ids after which it responds with a HTTP 400 error.
This affects the loading of non-workflow Resources that come after the StructureMaps as configured in the **Composition** resource 

The failing query is as below:
`https://ecbis-preview.smartregister.org/fhir/StructureMap?_id=97cf9bfb-90be-4661-8810-1c60be88f593,e4fcfd38-2f6b-4b58-af23-b1700f822b57,3c118304-fb18-4752-a296-941292811433,6a1748d8-6595-11ed-9022-0242ac120002,d65b7c69-209c-4b4b-95a4-32a968b8a21a,b4c2d528-5109-11ed-bdc3-0242ac120002,c7283c70-4f89-11ed-bdc3-0242ac120002,b4c2d528-5109-11ed-bdc3-0242ac120002,c7283c70-4f89-11ed-bdc3-0242ac120002,737a86d6-ad63-4345-8753-1e605438d1ee,98e45d1e-96f9-11ed-a8fc-0242ac120002,528a8603-2e43-4a2e-a33d-1ec2563ffd3e,0b7f0c7a-9b52-42bf-ab45-749918128a66,5667cfbd-13c4-4111-b952-7cee58bdb9d5,690b7313-ab5a-4c5f-8323-916c0025d804,febf0030-262f-443d-95b4-e002d437cfd0,ff7b22df-7645-43fe-9731-d90c3a21130b,253f460e-ef11-4af7-a2cc-80808280072f,74149320-97ce-463a-9cf3-206fad6e380f,ab62f8e8-e6d8-4ffb-b460-64b23e89b370,42d6d826-d6b2-41e9-9787-7424b4deca85,e71e7a6-194f-11ed-861d-0242ac120002,9fcb7a6a-4809-4f26-a379-c92bb783e5e0,2dc36363-a627-4fe6-b3c4-490f34389629,650943ec-3c61-42a7-86ef-12a4c9955d5d,ce46c251-b44e-4cb9-89aa-213c8ea47eb2,6256caaa-8781-4b34-985d-b20dcb41a2b3,ef19b62e-6cbf-43d6-beb5-4c0d9d2c20b8,63752b18-9f0e-48a7-9a21-d3714be6309a,befbdbcf-9013-4597-8f94-1a15e9f7f549,2ac0c108-c11d-44bf-b75f-fbf83072ec44,a92fbd36-baab-4670-b0bb-717f62072346,8f7828f5-3910-4bfc-94a9-a0da749fb37c`

### LINKED ISSUE #2535: [Quest] Incorrect Password Flagged as Internet Connection Issue
**Describe the bug**
When a user enters the wrong password during login, the App flags it as a possible internet connection issue. We need to ensure that the error for incorrect password can be differentiated from the error when there is no connection to ensure the right action is taken by the user. See sample screenshot below when I entered the wrong password.

**To Reproduce**
Enter incorrect password and tap on the login button.

**Expected behavior**
The App should give a different error when the password is incorrect and when there is no internet connection.

### PR DESCRIPTION
**Pull request by ndegwamartin**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

* Fixes Incorrect Password Flagged as Internet Connection Issue , Fixes #2535
* Enhance Config Sync - Implement Batch Syncing, Closes #2529

**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [x] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [x] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [x] I have verified any strings visible on UI components are in the `strings.xml` file
- [x] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [x] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by ndegwamartin** [comment]
Counterpart to 0.2.4 PR https://github.com/opensrp/fhircore/pull/2538

File: android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt [hunk 1]
```diff
@@ -417,5 +446,6 @@ constructor(
     const val TYPE_REFERENCE_DELIMITER = "/"
     const val CONFIG_SUFFIX = "_config"
     const val ICON_PREFIX = "ic_"
+    const val MANIFEST_PROCESSOR_BATCH_SIZE = 30
```
Comment (pld) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt:449]: do we need to optimize this?

**Review by pld — APPROVED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt (hunk 1 above)
Comment (ndegwamartin) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt:449]: what did you have in mind?

Comment (ndegwamartin) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt:449]: what did you have in mind for the optimization?

Comment (pld) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt:449]: nothing specific, I'm not sure if this is trivial and if we change this to 10 or 300 the performance won't be affected, or if it's something we can adjust to get performance gains, don't think we need to do anything for this PR, but if there's we think there's a >=50/50 chance we can get meaningful perf gains from adjusting this let's create an issue -- I didn't create an issue b/c I don't know if that's true, maybe any tweaks we do will be negligible to performance, what do you think?

Comment (dubdabasoduba) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt:449]: @ndegwamartin could we use a bundle transaction? So add all the items in a specific section of composition to a `transaction` bundle with GET requests methods. 

We would need to change the request from the app to POST.

Comment (dubdabasoduba) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt:449]: Kinda like what the SDK does while doing a sync-up.

File: android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt [hunk 2]
```diff
@@ -104,6 +104,26 @@ constructor(
     return decodedConfig
   }
 
+  inline fun <reified T : Configuration> retrieveConfigurations(configType: ConfigType): List<T> =
+    configsJsonMap.values
+      .filter {
+        try {
+          JSONObject(it).getString("configType").equals(configType.name, ignoreCase = true)
```
Comment (dubdabasoduba) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt:111]: Move the string to the constants

File: android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt [hunk 3]
```diff
@@ -343,36 +363,45 @@ constructor(
...
-            val resourceIds =
-              resourceGroup.value.joinToString(",") { sectionComponent ->
-                sectionComponent.focus.extractId()
-              }
-            val searchPath = resourceGroup.key + "?${Composition.SP_RES_ID}=$resourceIds"
+            val chunkedResourceIdList = resourceGroup.value.chunked(MANIFEST_PROCESSOR_BATCH_SIZE)
+
+            chunkedResourceIdList.forEach {
+              val resourceIds =
+                it.joinToString(",") { sectionComponent -> sectionComponent.focus.extractId() }
```
Comment (dubdabasoduba) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt:370]: move string to constant

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]
Some minor requests

**Review by pld — APPROVED** [review]

**Pull request merged**